packages/itwinui-react/src/core/Table/hooks/useScrollToRow.ts

```diff
@@ -54,8 +56,12 @@ export function useScrollToRow<T extends Record<string, unknown>>({
       return;
     }
 
-    rowRefs.current[pageRef.current[scrollToIndex]?.id]?.scrollIntoView();
-  }, [enableVirtualization, scrollToIndex]);
+    setTimeout(() => {
+      tableRef.current?.scrollTo({
```

Comment:
We cannot rely on `scrollTo`. It requires an explicit value, which is not always feasible to calculate. `offsetTop` might work for simple cases but will fail when the Table itself is not scrollable (e.g. when the page is scrollable and Table has `overflow: visible`).

Like I suggested earlier, let's use `scrollIntoView` and pick an approximate `scroll-margin` value that's bigger than the Table header (in its tallest wrapped view).

Reply:
It doesn't make sense to have two different implementations (three if you count virtualization), but I won't block the PR if you strongly before this. I just hope you understand that there is an implementation complexity to this, without any real UX benefit.

- Previously it was deemed acceptable that the row would be somewhere within the view, not necessarily at the top. So picking an approximate `scroll-margin` is not worse than before. It fixes the bug.
- Scrolling to the top might make sense to a developer, but the user might still be unaware of which row to look at (there's more than one row on the page).
- The highlight effect is a must for usable UX, and we should also recommend developers to move the focus (for keyboard users). Any solution we pick today must be in service of this end goal.

Other points to be aware of:
- Reading `offsetTop` forces [layout](https://gist.github.com/paulirish/5d52fb081b3570c81e3a), which could lead to a performance issue.
- Not having a buffer/padding around the scrolled row might cause an issue if we go with a highlight effect that extends outwards (e.g. if we use a box-shadow, it might get clipped by the Table header).

Reply:
Make sure the PR description is up-to-date.